lib/post_destroyer.rb

```diff
@@ -196,6 +198,16 @@ def trash_public_post_actions
     Post.with_deleted.where(id: @post.id).update_all(Hash[*f.flatten])
   end
 
+  def recover_public_post_actions(post_deleted_at)
+    PostAction.publics
+      .where(post_id: @post.id)
+      .with_deleted
+      .where(deleted_at: post_deleted_at..(post_deleted_at + 5.seconds))
```

Comment:
A small one but we usually try to avoid chaining multiple `where`. 

```
.where(
  post_id: @post.id,
  deleted_at: post_deleted_at..(post_deleted_at + 5.seconds)
)
```
is faster than

```
.where(post_id: @post.id)
.where(deleted_at: post_deleted_at..(post_deleted_at + 5.seconds))
```

Reply:
Why would combining them be more performant? I think ActiveRecord is smart enough to combine multiple `where` into a single SQL query - it doesn't hit the database for each one.

Reply:
O the SQL is the same. What I meant is that if you benchmark the Ruby code, it'll be faster